src/org/etools/j1939_84/bus/j1939/J1939DaRepository.java

```diff
@@ -160,11 +161,22 @@ static public void main(String... a) {
         List<PgnDefinition> pgns = new ArrayList<>(pgnLut.values());
         pgns.sort(Comparator.comparing(PgnDefinition::getId));
         for (PgnDefinition d : pgns) {
-            System.err.format("PGN: %6d: %6d %s%n", d.getId(), d.getBroadcastPeriod(), d.getLabel());
-            for (SpnDefinition s : d.getSpnDefinitions()) {
-                Slot slot = Slot.findSlot(s.getSlotNumber());
-                System.err.format("  SPN: %6d: %3d.%-3d %3d %6d %s%n", s.getSpnId(), s.getStartByte(), s.getStartBit(),
-                        slot != null ? slot.getLength() : -1, s.getSlotNumber(), s.getLabel());
+            if (d.getAcronym().startsWith("DM") && !J1939.isManual(d.getId())) {
+                System.err.format("PGN: %6d (%04X): %6d %s%n", d.getId(), d.getId(), d.getBroadcastPeriod(),
+                        d.getAcronym());
+                for (SpnDefinition s : d.getSpnDefinitions()) {
+                    int slotNumber = s.getSlotNumber();
+                    Slot slot = Slot.findSlot(slotNumber);
```

Comment:
Your changes need reconciled with 'trunk'.  I moved `findSlot` into `J1939DaRepository` as it didn't really belong in `Slot`.  In addition, I changed the logic never return a `null` Slot.